src/picongpu/include/particles/InterpolationForPusher.hpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2015 Richard Pausch
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+
+/** functor for particle field interpolator
+ *
+ * This functor is a simplification of the full
+ * field to particle interpolator that can be used in the
+ * particle pusher
+ */
+
+namespace picongpu
+{
+
+using namespace PMacc;
+
+
+template< typename T_Field2PartInt, typename T_MemoryType, typename T_FieldPosition >
+struct InterpolationForPusher
+{
+    typedef T_Field2PartInt Field2PartInt;
+
+    HDINLINE
+    InterpolationForPusher( const T_MemoryType& mem, const T_FieldPosition& fieldPos )
+        : m_mem( mem ), m_fieldPos( fieldPos )
+    {
+    }
+
+    template< typename T_PosType >
+    HDINLINE
+    float3_X operator()( const T_PosType& pos ) const
+    {
+        /* to do: memory shift, position shift */
```

Comment:
Note: How to implement the memory shift without register usage if not nedded:

Add a policy to this operator
```
template< typename T_PosType, typename ShiftPolicy >
HDINLINE
float3_X operator()( const T_PosType& pos , const ShiftPolicy = NoShift()) const
{
  return Field2PartInt()( 
    ShiftPolicy::memory(mem), 
    ShiftPolicy::position(pos), 
    m_fieldPos 
  );
}
```
NoShift::memory() - return the identity of the given memory
NoShift::position() - return the identity of the given position

By implementing a new policy like `ShiftToValidRange` the `operator()` can called like this:
```
...
 functorBField(pos,ShiftToValidRange());
...
```